app/controllers/spree/admin/taxjar_settings_controller.rb

```diff
@@ -41,6 +47,17 @@ def backfill_transactions
       def configuration_params
         params.require(:super_good_solidus_taxjar_configuration).permit(:preferred_reporting_enabled)
       end
+
+      def api_sync
+        if ENV["TAXJAR_API_KEY"]
```

Comment:
Is there any reason why we're guarding for this here 🤔 I know we have a similar check around the buttons in the view, but are we worried that someone is going to post directly to the endpoint?

I think sprinkling these checks throughout all our code is a bit of an anti-pattern. I would rather we let the request fail than keep adding checks like this.